proof/crefine/AARCH64/Machine_C.thy

```diff
@@ -0,0 +1,600 @@
+
+assumes enableFpuEL01_ccorres:
+  "ccorres dc xfdc \<top> UNIV []
+           (doMachineOp enableFpuEL01)
+           (Call enableFpuEL01_'proc)"
+
+assumes check_export_arch_timer_ccorres:
+    "ccorres dc xfdc \<top> UNIV []
+           (doMachineOp check_export_arch_timer)
+           (Call check_export_arch_timer_'proc)"
+
+assumes read_cntpct_ccorres:
+    "ccorres (=) ret__unsigned_longlong_' \<top> UNIV []
+           (doMachineOp read_cntpct)
+           (Call read_cntpct_'proc)"
+
+(* TLB and Cache ops *)
+
+assumes addressTranslateS1_ccorres:
+  "ccorres (=) ret__unsigned_long_' \<top> (\<lbrace>\<acute>vaddr___unsigned_long = vaddr \<rbrace>) []
+           (doMachineOp (addressTranslateS1 vaddr))
+           (Call addressTranslateS1_'proc)"
+
+assumes invalidateTranslationASID_ccorres:
+  "\<And>hw_asid. ccorres dc xfdc \<top> (\<lbrace>\<acute>hw_asid___unsigned_char = ucast hw_asid\<rbrace>) []
+                     (doMachineOp (invalidateTranslationASID hw_asid))
+                     (Call invalidateTranslationASID_'proc)"
+
+assumes invalidateTranslationSingle_ccorres:
+  "\<And>vptr. ccorres dc xfdc \<top> (\<lbrace>\<acute>vptr = vptr\<rbrace>) []
+                     (doMachineOp (invalidateTranslationSingle vptr))
+                     (Call invalidateTranslationSingle_'proc)"
+
+assumes cleanByVA_PoU_ccorres:
+  "ccorres dc xfdc \<top> (\<lbrace>\<acute>vaddr = w1\<rbrace> \<inter> \<lbrace>\<acute>paddr = w2\<rbrace>) []
+           (doMachineOp (cleanByVA_PoU w1 w2))
+           (Call cleanByVA_PoU_'proc)"
+
+assumes cleanCacheRange_RAM_ccorres:
+  "ccorres dc xfdc (\<lambda>s. w1 \<le> w2 \<and> w3 \<le> w3 + (w2 - w1)
+                        \<and> w1 && mask cacheLineSize  = w3 && mask cacheLineSize
+                        \<and> unat (w2 - w1) \<le> gsMaxObjectSize s)
+                   (\<lbrace>\<acute>start = w1\<rbrace> \<inter> \<lbrace>\<acute>end = w2\<rbrace> \<inter> \<lbrace>\<acute>pstart = w3\<rbrace>) []
+           (doMachineOp (cleanCacheRange_RAM w1 w2 w3))
+           (Call cleanCacheRange_RAM_'proc)"
+
+assumes cleanCacheRange_PoU_ccorres:
+  "ccorres dc xfdc (\<lambda>s. unat (w2 - w1) \<le> gsMaxObjectSize s
+                        \<and> w1 \<le> w2 \<and> w3 \<le> w3 + (w2 - w1)
+                        \<and> w1 && mask cacheLineSize = w3 && mask cacheLineSize)
+                   (\<lbrace>\<acute>start = w1\<rbrace> \<inter> \<lbrace>\<acute>end = w2\<rbrace> \<inter> \<lbrace>\<acute>pstart = w3\<rbrace>) []
+           (doMachineOp (cleanCacheRange_PoU w1 w2 w3))
+           (Call cleanCacheRange_PoU_'proc)"
+
+assumes cleanInvalidateCacheRange_RAM_ccorres:
+  "ccorres dc xfdc (\<lambda>s. unat (w2 - w1) \<le> gsMaxObjectSize s
+                        \<and> w1 \<le> w2 \<and> w3 \<le> w3 + (w2 - w1)
+                        \<and> w1 && mask cacheLineSize = w3 && mask cacheLineSize
+                        \<and> unat (w2 - w2) \<le> gsMaxObjectSize s)
```

Comment:
yes, it really should be. Thanks for catching that!

Reply:
Ok, this is slightly less bad than I was thinking. It's just a redundant clause, not masking a missing one. The assumption comes from the lemma with the same name on `ARM` (it's a lemma there, because the machine interface goes one deeper), and that lemma has the same redundant clause, which just disappears if you use the lemmas as `[simplified]` for instance. Since that one is a lemma, we can be sure that there is nothing missing or inconsistent, the only possibility is that it's redundant.

Will remove.

Reply:
Oh of course, I should have noticed that it was the same as the first assumption. It looks like similar lemmas are inconsistent with whether the equivalent of this is at the start or end of their assumptions, and my guess is that on ARM this was a copy-paste error which combined both options.

Reply:
Yes, that's probably what happened.